features/TESTS/filesystem/heap_block_device/main.cpp

```diff
@@ -78,30 +74,15 @@ void test_read_write() {
             write_block[i] = 0xff & rand();
         }
 
-        // erase, program, and read the block
-        printf("test  %0*llx:%llu...\n", addrwidth, block, block_size);
-
         err = bd.erase(block, block_size);
         TEST_ASSERT_EQUAL(0, err);
 
         err = bd.program(write_block, block, block_size);
         TEST_ASSERT_EQUAL(0, err);
 
-        printf("write %0*llx:%llu ", addrwidth, block, block_size);
-        for (int i = 0; i < 16; i++) {
-            printf("%02x", write_block[i]);
-        }
-        printf("...\n");
-
         err = bd.read(read_block, block, block_size);
         TEST_ASSERT_EQUAL(0, err);
 
-        printf("read  %0*llx:%llu ", addrwidth, block, block_size);
-        for (int i = 0; i < 16; i++) {
-            printf("%02x", read_block[i]);
-        }
-        printf("...\n");
```

Comment:
Uhm, wait a sec, these are used for determining the types of errors that occur during block read/write. Since this is the main test used for porting new block devices this is rather important.

Is pulling in printf and math really too much to run in CI?

If that's the case we really need to reconsider what sort of devices we are expecting to support storage. This test doesn't even have a filesystem yet!